app/models/user.rb

```diff
@@ -3,6 +3,9 @@ class User < ApplicationRecord
   has_many :comments, foreign_key: :author_id
   has_many :likes, foreign_key: :author_id
 
+  validates :name, presence: { message: 'Name can\'t be blank' }
+  validates :posts_counter, numericality: { greater_than_or_equal_to: 0, only_integer: true }
+
   def last_posts
     Post.where(author_id: id).order(created_at: :desc).limit(3)
   end
```

Comment:
- **[OPTIONAL]** We can refine this method. User has many posts as established by the association so the user has access to all the posts belonging to it in this model. Going by that we can refine this method so this:

  ```
  def last_posts
    posts.order(created_at: :desc).limit(3)
  end
  ```